Plot varbc coeffs

jcampinsp opened the pull request:
Here is pull resquest

jcampinsp committed:
Adding VarBC plotting scripts

jcampinsp committed:
Adding configuration file

ewhelan commented:
Thanks Joan. This will hopefully address #4

ewhelan committed:
Eoin's first comment

ewhelan committed:
Update 1_extract_varbc_cycle.sh to scan not loop

 * Use bash instead of ksh
 * Command line options added to ease of use
 * help option added
 * Scan an input directory instead of loop over an assumed directory
structure
 * Write output for all satellites and sensors

ewhelan commented:
Hola again.

I have taken a look at the "processing" script and made a few changes. Could you give it a whirl and let me know what you think.

Eoin

ewhelan committed:
Add script to installation and add fancy stuff

 * Add to cmake install
 * Have fun with echo statements!

jcampinsp commented:
Hi Eoin,

Of course, I'm going to look at your changes and test them on a set of 
VARBC.cycle files.
At first they seem fine to me, but later I will confirm it.

Joan

El 10/12/2021 a las 19:06, Eoin Whelan escribió:
>
> Hola again.
>
> I have taken a look at the "processing" script and made a few changes. 
> Could you give it a whirl and let me know what you think.
>
> Eoin
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-991183648>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AWRJW35FV7KNFNXD3D7LJALUQI6SVANCNFSM5JYUAUCA>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>
>
> [ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", 
> "potentialAction": { ***@***.***": "ViewAction", "target": 
> "https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-991183648", 
> "url": 
> "https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-991183648", 
> "name": "View Pull Request" }, "description": "View this Pull Request 
> on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", 
> "url": "https://github.com" } } ] 
-- 
Joan Campins Pons
[Logotipo de Aemet]
Delegación Territorial en Illes Balears
Técnico Superior Estudios Meteorológicos Mediterráneo
Teléfonos: Interno: 50791 Externo: (+34)971 145410
Correo Electrónico: ***@***.***
www.aemet.es <http://www.aemet.es>
<https://www.facebook.com/AgenciaEstatalMeteorologia/> 
<https://twitter.com/AEMET_Baleares/> 
<https://www.youtube.com/channel/UCd-ceYPisAtCmmoZa26I-5g> 
<https://aemetblog.es/>

ewhelan commented:
>   Only if there is a easy way to do:
> * Write output for all satellites and sensors: may be the default can be write output for all, but select for satellite/sensor could be an option
> * Similarly, DTGBEG and DTGEND can be an option, but the default is all files in input_dir
 OK. These sound like nice features. I can work on this assuming the Harmonie archive structure. I used the scan/find to cater for other directory structures (outside of Harmonie).

ewhelan committed:
Add option to select sat/sensors on CLI

Add command line options to select satellites (-S) and sensors (-s) for
processing.

ewhelan committed:
Start beautifying the plotting script

Add CLI options.

ewhelan committed:
Rename plotting script and add to install

ewhelan committed:
And get the plotting working

 * I have probably broken some of the R time-series functionality
 * I don't think we need include.paths or mandtg anymore

ewhelan commented:
More updates @jcampinsp 

With these changes I am producing plots but I need to tidy ...

Would be great if you could test.

jcampinsp commented:
Tomorrow I will test all these changes, hopefully only a few changes 
will be necessary to produce the same plots I used to obtain with old 
scripts.

Joan


El 13/12/2021 a las 19:54, Eoin Whelan escribió:
>
> More updates @jcampinsp <https://github.com/jcampinsp>
>
> With these changes I am producing plots but I need to tidy ...
>
> Would be great if you could test.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-992771594>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AWRJW35J4II2U57KO5GJIZLUQY6PRANCNFSM5JYUAUCA>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>
>
> [ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", 
> "potentialAction": { ***@***.***": "ViewAction", "target": 
> "https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-992771594", 
> "url": 
> "https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-992771594", 
> "name": "View Pull Request" }, "description": "View this Pull Request 
> on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", 
> "url": "https://github.com" } } ] 
-- 
Joan Campins Pons
[Logotipo de Aemet]
Delegación Territorial en Illes Balears
Técnico Superior Estudios Meteorológicos Mediterráneo
Teléfonos: Interno: 50791 Externo: (+34)971 145410
Correo Electrónico: ***@***.***
www.aemet.es <http://www.aemet.es>
<https://www.facebook.com/AgenciaEstatalMeteorologia/> 
<https://twitter.com/AEMET_Baleares/> 
<https://www.youtube.com/channel/UCd-ceYPisAtCmmoZa26I-5g> 
<https://aemetblog.es/>

jcampinsp committed:
Update 2_varbc_fig.sh

VARBCOUT is the output directory, where png files will be stored:
1) one-line files are in VARBCINP but the are attached (with cat) in VARBCOUT
2) script varbc_plot.Rsh read the ascci file in VARBCOUT and write the png file also in VARBCOUT
3) the varbc_plot.Rsh has to include indir into file.png

jcampinsp committed:
Update varbc_plot.Rsh

included indir in file.png

ewhelan committed:
Add (Python) VarBC ts plotter

ewhelan commented:
Hola @jcampinsp 

More tweaking from me. A change to the output format in 1_extract_varbc_cycle and I have added a Python based plotter. Let me know what you think:

1. Install
```
cd /path/to/clone
mkdir -p build && cd build
cmake .. -DCMAKE_INSTALL_PREFIX=$HOME/metapp/da_tools && make && ctest && make install
export PATH=$HOME/metapp/da_tools/bin:$PATH
```
2. Test
```
cd /path/to/test_dir
1_extract_varbc_cycle -i /path/to/varbc/input_dir/ -o output_dir
plotvarbcpred -i oper_latest/VARBC_5_3_9_210000 -b && eog varbc_pred_5_3_9.png
```

jcampinsp commented:
Hello Eoin,

I succeed to install the scripts  1_extract_varbc_cycle.sh and 
plotvarbcpred.py

I run 1_extract_varbc_cycle, and output files were created, but when 
running  plotvarbcpred I get an error:

noreste:/spred/pn19/github/jcampinsp/PlotVarbcCoeffs > plotvarbcpred -i 
./tests/EXP_SEVIRI/aux2/VARBC_70_29_2 -b
Read: ./tests/EXP_SEVIRI/aux2/VARBC_70_29_2
Traceback (most recent call last):
   File "/spred/pn19/metapp/da_tools/bin/plotvarbcpred", line 229, in 
<module>
     sys.exit(main(sys.argv))
   File "/spred/pn19/metapp/da_tools/bin/plotvarbcpred", line 226, in main
plot_varbc_pred_ts(tsdatetime,tsdata,args.labels,args.lloc,args.b)
   File "/spred/pn19/metapp/da_tools/bin/plotvarbcpred", line 129, in 
plot_varbc_pred_ts
     title_string='VarBC Predictors for '+sat_map[nsat]+': ' 
+sen_map[nsen]+ ' Channel '+str(nchn)
KeyError: 29

Did I something wrong (when installing) or more arguments are needed?


Joan

El 21/12/2021 a las 10:00, Eoin Whelan escribió:
>
> Hola @jcampinsp <https://github.com/jcampinsp>
>
> More tweaking from me. A change to the output format in 
> 1_extract_varbc_cycle and I have added a Python based plotter. Let me 
> know what you think:
>
>  1. Install
>
> |cd /path/to/clone mkdir -p build && cd build cmake .. 
> -DCMAKE_INSTALL_PREFIX=$HOME/metapp/da_tools && make && ctest && make 
> install export PATH=$HOME/metapp/da_tools/bin:$PATH |
>
>  2. Test
>
> |cd /path/to/test_dir 1_extract_varbc_cycle -i 
> /path/to/varbc/input_dir/ -o output_dir plotvarbcpred -i 
> oper_latest/VARBC_5_3_9_210000 -b && eog varbc_pred_5_3_9.png |
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-998592425>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AWRJW35L6NSXF5FQ7YAHUJDUSA623ANCNFSM5JYUAUCA>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>
> [ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", 
> "potentialAction": { ***@***.***": "ViewAction", "target": 
> "https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-998592425", 
> "url": 
> "https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-998592425", 
> "name": "View Pull Request" }, "description": "View this Pull Request 
> on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", 
> "url": "https://github.com" } } ] 
-- 
Joan Campins Pons
[Logotipo de Aemet]
Delegación Territorial en Illes Balears
Técnico Superior Estudios Meteorológicos Mediterráneo
Teléfonos: Interno: 50791 Externo: (+34)971 145410
Correo Electrónico: ***@***.***
www.aemet.es <http://www.aemet.es>
<https://www.facebook.com/AgenciaEstatalMeteorologia/> 
<https://twitter.com/AEMET_Baleares/> 
<https://www.youtube.com/channel/UCd-ceYPisAtCmmoZa26I-5g> 
<https://aemetblog.es/>

ewhelan commented:
This is related to SEVIRI sensor entry in sen_map (incorrectly 70). Should it be 29?

On [https://apps.ecmwf.int/odbgov/sensor/ ](https://apps.ecmwf.int/odbgov/sensor/) it is 21. Could you mail me your test input file. Thanks.

ewhelan committed:
Correct SEVIRI sensor value in map array

ewhelan commented:
Set to 29 (as it is in Bator). Could you repeat your test. Thanks.

jcampinsp commented:
I set to 29 and it works. I created a png file!. Thanks Eoin.

Do you think I have to check varbc_plot.Rsh to have two plotting options 
or with one is enough?

Joan


El 21/12/2021 a las 13:45, Eoin Whelan escribió:
>
> Set to 29 (as it is in Bator). Could you repeat your test. Thanks.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-998749433>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AWRJW3325J7MUT7RMTPUEKTUSBZF7ANCNFSM5JYUAUCA>.
> Triage notifications on the go with GitHub Mobile for iOS 
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> 
> or Android 
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>. 
>
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>
> [ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", 
> "potentialAction": { ***@***.***": "ViewAction", "target": 
> "https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-998749433", 
> "url": 
> "https://github.com/Hirlam/AccordDaTools/pull/9#issuecomment-998749433", 
> "name": "View Pull Request" }, "description": "View this Pull Request 
> on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", 
> "url": "https://github.com" } } ] 
-- 
Joan Campins Pons
[Logotipo de Aemet]
Delegación Territorial en Illes Balears
Técnico Superior Estudios Meteorológicos Mediterráneo
Teléfonos: Interno: 50791 Externo: (+34)971 145410
Correo Electrónico: ***@***.***
www.aemet.es <http://www.aemet.es>
<https://www.facebook.com/AgenciaEstatalMeteorologia/> 
<https://twitter.com/AEMET_Baleares/> 
<https://www.youtube.com/channel/UCd-ceYPisAtCmmoZa26I-5g> 
<https://aemetblog.es/>

ewhelan committed:
Update sensor map entries

ewhelan committed:
Remove scripts/2_varbc_fig.sh&share/include.paths

No longer needed.

ewhelan committed:
Rename varbc diagnose script

ewhelan committed:
Add some simple tests

ewhelan committed:
Update README for VarBC tools

ewhelan commented:
Think we are good to go with this one.

ewhelan reviewed (approved):
Think we are both happy now with this PR.
